wallets/provider-brave/src/signer-solana.ts

```diff
@@ -0,0 +1,37 @@
+import { executeSolanaTransaction } from '@rango-dev/signer-solana';
+import {
+  type GenericSigner,
+  SignerError,
+  SignerErrorCode,
+  type SolanaTransaction,
+} from 'rango-types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type SolanaExternalProvider = any;
+
+export class CustomSolanaSigner implements GenericSigner<SolanaTransaction> {
+  private provider: SolanaExternalProvider;
+  constructor(provider: SolanaExternalProvider) {
+    this.provider = provider;
+  }
+
+  async signMessage(msg: string): Promise<string> {
+    try {
+      const encodedMessage = new TextEncoder().encode(msg);
+      const { signature } = await this.provider.request({
+        method: 'signMessage',
+        params: {
+          message: encodedMessage,
+        },
+      });
+      return signature;
+    } catch (error) {
+      throw new SignerError(SignerErrorCode.SIGN_TX_ERROR, undefined, error);
+    }
+  }
+
+  async signAndSendTx(tx: SolanaTransaction): Promise<{ hash: string }> {
+    const hash = await executeSolanaTransaction(tx, this.provider);
+    return { hash };
+  }
+}
```

Comment:
Please refactor your duplicate changes by extending `DefaultSolanaSigner` as we've talked offline.